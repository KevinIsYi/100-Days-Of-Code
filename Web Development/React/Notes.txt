{Go to folder}
npx create-react-app `name`
cd `name`
npm start

rafcp 
import React from 'react'
import PropTypes from 'prop-types'

const firstApp = props => {
    return (
        <div>
            
        </div>
    )
}

firstApp.propTypes = {

}

export default firstApp