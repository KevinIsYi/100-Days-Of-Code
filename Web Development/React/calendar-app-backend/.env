PORT=4000
DB_CONNECTION=mongodb+srv://mern_user:NekxHA9lixWzsT0c@cluster0.9xvmk.mongodb.net/mern_calendar
JWT_SEED=p@L@BR@_S3CR3T@_JWT