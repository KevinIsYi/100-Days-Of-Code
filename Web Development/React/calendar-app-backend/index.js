const express = require('express');
const dotenv = require('dotenv');
const cors = require('cors');
const { dbConnection } = require('./database/config');

dotenv.config();

// Create express server
const app = express();

// Database
dbConnection();


//CORS
app.use(cors());

app.use(express.static('public'));

// Lecture and parse of body
app.use( express.json() );

app.use('/api/auth', require('./routes/auth'));
app.use('/api/events', require('./routes/events'));


// Listen petitions
app.listen(process.env.PORT, () => { // Port, callback
    console.log(`Server is running on: ${ process.env.PORT }`);
});