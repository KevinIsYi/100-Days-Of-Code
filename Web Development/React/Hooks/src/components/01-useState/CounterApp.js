import React, { useState } from 'react'
import './counter.css';

export const CounterApp = () => {

    const [state, setState] = useState({
        counter1: 0,
        counter2: 0,
        counter3: 10 
    });

    const {counter1, counter2} = state;

    const handleClick = () => {
        setState({
            ...state,
            counter1: counter1 + 1
        });
    };

    return (
        <>
            <h1>Counter 1: { counter1 }</h1>
            <h1>Counter 2: { counter2 }</h1>
            <button className="btn btn-primary" onClick={handleClick}>+1</button>   
        </>
    )
}
