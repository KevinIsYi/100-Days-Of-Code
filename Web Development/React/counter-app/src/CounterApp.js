import React, {useState} from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({value}) => {

    const [ counter, setCounter] = useState(value);
    console.log(counter);

    const sum = (e) => {
        setCounter(counter + 1);
        //setCounter((c) => c + 1);
    }
    const reset = (e) => {
        setCounter(value);
    }
    const rest = (e) => {
        setCounter(counter - 1);
    }
    return (
        <>
            <h1>Counter App</h1>
            <h2>{counter}</h2>

            <button onClick={sum}>+1</button>
            <button onClick={reset}>RESET</button>
            <button onClick={rest}>-1</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;