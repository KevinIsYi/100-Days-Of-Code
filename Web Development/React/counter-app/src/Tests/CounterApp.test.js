import React from 'react';
import "@testing-library/jest-dom";
import CounterApp from '../CounterApp';
import { shallow } from "enzyme";

describe("counterApp test", () => {
    test("Must show <CounterApp/> correctly", () => {
        const wrapper = shallow(<CounterApp/>);
        expect(wrapper).toMatchSnapshot();
    });
    test("Must show default value", () => {
        const value = 100;
        const wrapper = shallow(<CounterApp value={value}/>); 
        const fromSnapShot = wrapper.find("h2").text();

        expect(value).toBe(value);
    });
});