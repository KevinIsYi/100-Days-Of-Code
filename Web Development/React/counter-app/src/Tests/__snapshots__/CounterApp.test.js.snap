// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counterApp test Must be equal to snapshot 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    RESET
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;

exports[`counterApp test Must show <CounterApp/> correctly 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    RESET
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
